packages/lib/zuauth/src/server.ts

```diff
@@ -52,17 +55,27 @@ export async function authenticate(
       isEqualEdDSAPublicKey(pubKey, pcd.claim.signer)
     )
   ) {
-    throw new Error(
+    throw new ZuAuthAuthenticationError(
       "Signing key does not match any of the configured public keys"
     );
   }
 
+  if (
+    eventIds.size > 0 &&
+    pcd.claim.partialTicket.eventId &&
```

Comment:
I think you should also check that either eventID is revealed, or validEventIDs is used and checked (see below), and throw if neither is true.  Unless you think there's a case where the user intentionally doesn't want the eventID checked (in which case why would it be in the config?).

My thought process which leads to this:
One could imagine in general checking that all the fields in `partialTicket` which should be revealed are actually revealed.  You want to avoid cases where the remote site decided to give you a proof which doesn't reveal enough data.

Probably in the general ZuAuth case that's up to the 3rd-party code using the library, which will presumably notice if any field they depend on is missing.  However for fields checked by ZuAuth (eventID, productID) the app code might reasonably assume that it doesn't need to bother looking at those fields, since ZuAuth already checked them.

A case I'm a bit worried about is this scenario:
- Proof config has >20 event IDs so no use of validEventIds
- eventID is configured to be revealed
- Malicious remote app generates a proof with eventID not revealed
- This code doesn't check anything
- App code doesn't make use of event ID (assuming that ZuAuth has verified it) and never notices.